staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -383,29 +383,35 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 		server, requestGetter := newHTTP2ServerWithClient(handler, requestTimeout*2)
 		defer server.Close()
 
-		var err error
-		_, err = requestGetter(firstRequestPathPanic)
-		if !executed {
-			t.Errorf("Expected inner handler to be executed for request: %q", firstRequestPathPanic)
-		}
-		if isClientTimeout(err) {
-			t.Fatalf("the client has unexpectedly timed out - request: %q error: %s", firstRequestPathPanic, err.Error())
-		}
-		expectResetStreamError(t, err)
+		// we send two requests synchronously, one at a time
+		//  - first request is expected to panic as designed
+		//  - second request is expected to success
+		func() {
```

Comment:
I don't understand what's the point of opaqueing these into func() { ... }() - can you explain?